dipy/reconst/dki_micro.py

```diff
@@ -0,0 +1,489 @@
+        maximal value of kurtosis.
+    gtol : float, optional
+        This input is to refine kurtosis maxima under the precision of the
+        directions sampled on the sphere class instance. The gradient of the
+        convergence procedure must be less than gtol before successful
+        termination. If gtol is None, fiber direction is directly taken from
+        the initial sampled directions of the given sphere object
+    mask : ndarray
+        A boolean array used to mark the coordinates in the data that should be
+        analyzed that has the shape dki_params.shape[:-1]
+
+    Returns
+    --------
+    AWF : ndarray (x, y, z) or (n)
+        Axonal Water Fraction
+
+    References
+    ----------
+    .. [1] Fieremans E, Jensen JH, Helpern JA, 2011. White matter
+           characterization with diffusional kurtosis imaging.
+           Neuroimage 58(1):177-88. doi: 10.1016/j.neuroimage.2011.06.006
+    """
+    kt_max = kurtosis_maximum(dki_params, sphere=sphere, gtol=gtol, mask=mask)
+
+    AWF = kt_max / (kt_max + 3)
+
+    return AWF
+
+
+def diffusion_components(dki_params, sphere='symmetric362', awf=None,
+                         mask=None):
+    """ Extracts the intra and extra-cellular diffusion tensors of well aligned
+    fibers from diffusion kurtosis imaging parameters [1]_.
+
+    Parameters
+    ----------
+    dki_params : ndarray (x, y, z, 27) or (n, 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+    sphere : Sphere class instance, optional
+        The sphere providing sample directions to sample the intra and extra
+        cellular diffusion tensors. For more details see Fieremans et al.,
+        2011.
+    awk : ndarray (optional)
+        Array containing values of the axonal water fraction that has the shape
+        dki_params.shape[:-1]. If not given this will be automatically computed
+        using function axonal_water_fraction with function's default precision.
+    mask : ndarray (optional)
+        A boolean array used to mark the coordinates in the data that should be
+        analyzed that has the shape dki_params.shape[:-1]
+
+    Returns
+    --------
+    EDT : ndarray (x, y, z, 12) or (n, 12)
+        Parameters of the extra-cellular diffusion tensor.
+    IDT : ndarray (x, y, z, 12) or (n, 12)
+        Parameters of the intra-cellular diffusion tensor.
+
+    Notes
+    -----
+    The parameters of both extra-cellular and intra-cellular diffusion tensors
+    are order as follow:
+        1) Three diffusion tensor's eingenvalues
+        2) Three lines of the eigenvector matrix each containing the first,
```

Comment:
Exactly - we only need the eigenvalues of the diffusion tensor to compute the standard diffusion and kurtosis measures that we imported from class inheritance. That is also why I am not decomposing the hindered and restricted diffusion tensors.